### PR TITLE
Added user guide/documentation and unit testing to the vision pipeline

### PR DESCRIPTION
**Pull request by cat101**

### CODE REVIEW

File: landingai/predict.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def _load_api_credential(
         """Create an APICredential object from the provided api_key and api_secret if they are not none.
         Otherwise, it will try to load from the environment variables or .env file.
         """
-        if api_key is None or api_secret is None:
+        if api_key is None:  # api_secret is now optional
```
Comment (AsiaCao) [inline landingai/predict.py:65]: Nice!
we need some documentation along with this change, probably need to explain in what circumstance the api secret is optional, i.e. v1 keys can't be optional. maybe also explain a bit about API key v1 and v2.
I was also thinking maybe it's less error prone and clearer (in terms of naming) if we create a new V2 class for it.

File: tests/unit/landingai/test_vision_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import cv2
+import numpy as np
+
+from landingai.vision_pipeline import NetworkedCamera
+
+
+def test_NetworkedCamera():
```
Comment (AsiaCao) [inline tests/unit/landingai/test_vision_pipeline.py:7]: nit about naming convention: snake case doesn't use upper case in a single word. so it has to be either `test_networkedcamera` or `test_networked_camera`

File: docs/user_guide/1_concepts.md [hunk 3]
```diff
@@ -16,32 +16,40 @@ This section explains important high-level concepts that will help you better us
...
 
-Before using this library for inference, you need train your model in `LandingLens` and [deploy it](https://support.landing.ai/docs/deployment-options).
+Before using this library for inference, you need train your model in LandingLens and [deploy it](https://support.landing.ai/docs/deployment-options).
 
 This library supports two deployment options:
 - [Cloud Deployment](https://support.landing.ai/landinglens/docs/cloud-deployment)
 - LandingEdge (support coming soon...)
 
-The easiest way to get started is using `CloudDeployment`.
+The easiest way to get started is using A Cloud Deployment.
```
Comment (AsiaCao) [inline docs/user_guide/1_concepts.md:25]: nit, grammar: "A Cloud Deployment" -> "a Cloud Deployment"?

File: docs/user_guide/1_concepts.md [hunk 4]
```diff
@@ -16,32 +16,40 @@ This section explains important high-level concepts that will help you better us
...
 - Segmentation: `landingai.common.SegmentationPrediction`
 - Visual Prompting: `landingai.common.SegmentationPrediction`
 
 This library also defines a parent class, `landingai.common.Prediction`, which represent a generic prediction that contains all the shared attributes across all of the subclasses.
+
+### Building sophisticated vision pipelines
+
+Real world problems require a multi step approach in order to get results. For example: if we want to count the number and model of cars going over a bridge, we will need to take images from a live RTSP stream and act when motion is detected. We will also need to use object detection to identify individual cars and track them as they traverse the frame. We will then need to classify the cars and collect statistics to product the desired report.
+
+While this image processing pipeline may seem daunting, it has a many commonalities with other vision problems. This is why we developed a vision pipeline concept where `Frames` produced by an image source can be passed across processing modules using method chaining.
```
Comment (AsiaCao) [inline docs/user_guide/1_concepts.md:53]: "it has a many " -> "it has many "

File: docs/user_guide/4_pipelines.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+## Vision Pipelines
+
+Pipelines can simplify complex vision tasks by breaking it into a sequence of operations that are applied to every image. Images are modeled as `landingai.vision_pipeline.Frame` and sequences of images as `landingai.vision_pipeline.FrameSet`. Most image sources will produce a `FrameSet` even when it contains a single image (e.g. `landingai.vision_pipeline.NetworkedCamera`). This is convenient as it allows subsequent stages of the pipelines to introduce new derived frames as part of the processing (e.g. when extracting faces from an initial Frame into new Frames).
```
Comment (AsiaCao) [inline docs/user_guide/4_pipelines.md:3]: grammar/typo: "sequences of images as `landingai.vision_pipeline.FrameSet`" -> "sequences of images is modeled as `landingai.vision_pipeline.FrameSet`"

It'd be helpful to expand on this topic: "Most image sources will produce a `FrameSet` even when it contains a single image" 
I think some additional context to share would be: `NetworkedCamera` is an iterator of captured frame. Each frame/image is modeled as an `FrameSet` (not a Frame), though there is only one image in the `FrameSet`.
Adding an example is even better.

File: docs/user_guide/4_pipelines.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+The following example shows a basic pipeline starting from a single image:
+```python
+FrameSet.fromImage("sample_images/1196.png")
+    .run_predict(predictor=...)
+    .overlay_predictions()
+    .show_image() 
+    .show_image(image_src="overlay") 
+```
```
Comment (AsiaCao) [inline docs/user_guide/4_pipelines.md:12]: it'd be helpful to explain the expected behavior of this example.
in particular, as a new developer, i'd be curious about what does `show_image()` do? why there are two `show_image`? what's "image_src" paramater and where does "overlay" come from?

**Review by AsiaCao — APPROVED** [review]
nice

File: docs/user_guide/1_concepts.md [hunk 7]
```diff
@@ -16,32 +16,40 @@ This section explains important high-level concepts that will help you better us
...
 - Classification:`landingai.common.ClassificationPrediction`
 - Object Detection:`landingai.common.ObjectDetectionPrediction`
 - Segmentation: `landingai.common.SegmentationPrediction`
 - Visual Prompting: `landingai.common.SegmentationPrediction`
 
 This library also defines a parent class, `landingai.common.Prediction`, which represent a generic prediction that contains all the shared attributes across all of the subclasses.
+
+### Building sophisticated vision pipelines
+
+Real world problems require a multi step approach in order to get results. For example: if we want to count the number and model of cars going over a bridge, we will need to take images from a live RTSP stream and act when motion is detected. We will also need to use object detection to identify individual cars and track them as they traverse the frame. We will then need to classify the cars and collect statistics to product the desired report.
```
[the remark this answers is not in the document]
Comment (cat101) [reply docs/user_guide/1_concepts.md:51]: Done

File: docs/user_guide/1_concepts.md (hunk 3 above)
Comment (cat101) [reply docs/user_guide/1_concepts.md:25]: done

File: docs/user_guide/1_concepts.md (hunk 4 above)
Comment (cat101) [reply docs/user_guide/1_concepts.md:53]: done

File: docs/user_guide/4_pipelines.md (hunk 6 above)
Comment (cat101) [reply docs/user_guide/4_pipelines.md:12]: done

File: landingai/predict.py (hunk 1 above)
Comment (cat101) [reply landingai/predict.py]: Reverted

File: tests/unit/landingai/test_vision_pipeline.py (hunk 2 above)
Comment (cat101) [reply tests/unit/landingai/test_vision_pipeline.py]: done

**Pull request merged**